Comma separated amount

KarishmaBothara opened the pull request.

benxgao commented:
Normally, we need sufficient unit test coverage on this kind of algorithm calculating, to cover all edge situations. 

Is there any existing currency formatting JS libs that we can borrow here, so that we do not need write and test algorithms by our own if it is already done well by others?

Also a JS lib may already have resolved most problems in the currency area, and we just need to leverage it to deal with potential requirements on currency formatting in the future?

KarishmaBothara commented:
> Normally, we need sufficient unit test coverage on this kind of algorithm calculating, to cover all edge situations.
> 
> Is there any existing currency formatting JS libs that we can borrow here, so that we do not need write and test algorithms by our own if it is already done well by others?
> 
> Also a JS lib may already have resolved most problems in the currency area, and we just need to leverage it to deal with potential requirements on currency formatting in the future?

Thanks for your comment..I have updated the PR and used the polkadot's existing utility which they have used through out the app to format the balance. Have removed the custom function that I added which replaced the number with comma separated one using regex..

benxgao commented on packages/react-components/src/InputNumber.tsx:
```diff
@@ -231,8 +241,7 @@ export default function InputNumber (props: Props): React.ReactElement<Props> {
       const newValue = `${value.substring(0, i || 0)}${event.key}${value.substring(j || 0)}`;
-
-      if (!getRegex(isDecimal || !!si).test(newValue)) {
+      if (!getRegex(isDecimal || !!si).test(newValue.replace(/,/g, ''))) {
```
Is there any other places could possible use this format?
